crates/ruff_python_parser/src/string.rs

```diff
@@ -605,22 +324,31 @@ pub(crate) fn parse_strings(
                 implicit_concatenated,
             }),
             kind: None,
-            range: TextRange::new(initial_start, last_end),
+            range,
         }
         .into());
     }
 
     if !has_fstring {
         let mut content: Vec<String> = vec![];
```

Comment:
We should change this to a `String` and use `push_str` rather than building a vec first and than joining the parts.